src/Core/Components/MessageBar/FluentMessageBarContainer.razor.cs

```diff
@@ -83,19 +83,29 @@ protected IEnumerable<Message> MessagesToShow
     /// <summary />
     protected override void OnInitialized()
     {
-        MessageService.OnMessageItemsUpdated += OnAlertUpdateHandler;
+        MessageService.OnMessageItemsUpdated += OnMessageItemsUpdatedHandler;
+        MessageService.OnMessageItemsUpdatedAsync += OnMessageItemsUpdatedHandlerAsync;
         base.OnInitialized();
     }
 
     /// <summary />
-    protected virtual void OnAlertUpdateHandler()
+    protected virtual void OnMessageItemsUpdatedHandler()
     {
         InvokeAsync(StateHasChanged); 
     }
 
+    protected async virtual Task OnMessageItemsUpdatedHandlerAsync()
+    {
+        await Task.Run(() =>
+        {
+            InvokeAsync(StateHasChanged);
+        });
+    }
+
     /// <summary />
     public void Dispose()
     {
-        MessageService.OnMessageItemsUpdated -= OnAlertUpdateHandler;
+        MessageService.OnMessageItemsUpdated -= OnMessageItemsUpdatedHandler;
+        MessageService.OnMessageItemsUpdatedAsync -= OnMessageItemsUpdatedHandlerAsync;
```

Comment:
I think you should replace `!` by `?` on `await OnMessageItemsUpdatedAsync!.Invoke();`, in `MessageService` class (line 222)

Reply:
No, can't change because of `await`